valhalla/meili/geometry_helpers.h

```diff
@@ -18,24 +19,23 @@ namespace helpers {
 // snapped point, sqaured distance, segment index, offset
 inline std::tuple<midgard::PointLL, float, typename std::vector<midgard::PointLL>::size_type, float>
 Project(const midgard::projector_t& p,
-        const typename std::vector<midgard::PointLL>& shape,
+        midgard::Shape7Decoder<midgard::PointLL>& shape,
         float snap_distance = 0.f) {
-  if (shape.empty()) {
-    throw std::invalid_argument("got empty shape");
-  }
-
-  midgard::PointLL closest_point(shape.front());
+  midgard::PointLL first_point(shape.pop());
+  auto closest_point = first_point;
+  auto closest_segment_point = first_point;
```

Comment:
we have to do more book keeping though because we cant just index into the full shape